src/main/default/lwc/abandonRequestButton/__tests__/abandonRequestButton.test.js

```diff
@@ -0,0 +1,128 @@
+  }
+);
+
+describe('c-abandon-request-button', () => {
+  afterEach(() => {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+  });
+
+  it('should display the abandon request button', async () => {
+    const element = createTestComponent();
+
+    await allPromisesResolution();
+    const buttonNode = element.shadowRoot.querySelector('button');
+    const labelNode = element.shadowRoot.querySelector('label');
+
+    expect(buttonNode).not.toBeNull();
+    expect(labelNode).toBeNull();
+    await expect(element).toBeAccessible();
+  });
+
+  it('should display the correct button label', async () => {
+    const element = createTestComponent();
+    const expectedText = 'Expected Text';
+    element.buttonLabel = expectedText;
+
+    await allPromisesResolution();
+    const buttonNode = element.shadowRoot.querySelector('button');
+
+    expect(buttonNode).not.toBeNull();
+    expect(buttonNode.textContent).toBe(expectedText);
+    await expect(element).toBeAccessible();
+  });
+
+  it('should display the correct default button label', async () => {
+    const element = createTestComponent();
+    const expectedText = `Abandon Request`;
+
+    await allPromisesResolution();
+    const buttonNode = element.shadowRoot.querySelector('button');
+
+    expect(buttonNode).not.toBeNull();
+    expect(buttonNode.textContent).toBe(expectedText);
+    await expect(element).toBeAccessible();
+  });
+
+  it('should display the correct label', async () => {
+    const element = createTestComponent();
+    const expectedText = 'Expected Text';
+    element.label = expectedText;
+
+    await allPromisesResolution();
+    const labelNode = element.shadowRoot.querySelector('label');
+
+    expect(labelNode).not.toBeNull();
+    expect(labelNode.textContent).toBe(expectedText);
+    await expect(element).toBeAccessible();
+  });
+
+  it('should display a big button when size is set to big', async () => {
+    const element = createTestComponent();
+    const bigButtonClass = 'big';
+    element.size = 'big';
+
+    await allPromisesResolution();
+    const buttonNode = element.shadowRoot.querySelector('button');
+
+    expect(buttonNode).not.toBeNull();
+    expect(buttonNode.classList.contains(bigButtonClass)).toBe(true);
+    await expect(element).toBeAccessible();
+  });
+
+  it('should display a small button when size is set to small', async () => {
+    const element = createTestComponent();
+    const bigButtonClass = 'big';
+    element.size = 'small';
+
+    await allPromisesResolution();
+    const buttonNode = element.shadowRoot.querySelector('button');
+
+    expect(buttonNode).not.toBeNull();
+    expect(buttonNode.classList.contains(bigButtonClass)).toBe(false);
+    await expect(element).toBeAccessible();
+  });
+
+  it('should open the abandon modal when clicking on the button', async () => {
```

Comment:
Is it possible to also validate that the proper Headless action is dispatched?

Reply:
I think you should be able to mock `CoveoHeadlessCaseAssist.loadCaseAssistAnalyticsActions(engine)` with jest and add an `expect(mock).toHaveBeenCalledTimes(1)`;

Reply:
I'll try this👌

Reply:
I don't think it would be possible to validate that the proper Headless actions are dispatched at this level since we need to wrap the component with a quanticCaseAssistInterface to be able to see the Headless actions working correctly, and without having a quanticCaseAssistInterface we don't even have the CoveoHeadlessCaseAssist object. so I don't think we can include such tests in the unit tests.